clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -4942,4 +4862,8 @@ KafkaDataIntegrityValidator getDataIntegrityValidator() {
   long getLocalVtSubscribeOffset(PartitionConsumptionState pcs) {
     return (isGlobalRtDivEnabled()) ? pcs.getLatestConsumedVtOffset() : pcs.getLatestProcessedLocalVersionTopicOffset();
   }
+
+  HeartbeatMonitoringService getHeartbeatMonitoringService() {
```

Comment:
Hrm.  Idk about this.  I know 'practically' speaking this is going to work.  But conceptually, a heartbeat is a notion of a leader which creates heartbeats.  Additionally, the LeaderFollower SIT is tied to the LeaderFollower statemodel (same as the heartbeat service).  Do we really need it here?

Reply:
It could be that just via code rot StoreIngestionTask has already compromised the isolation of this notion.  idk

Reply:
Yes, I think today it is bad. I want to move this whole ready-to-serve thing out - or the lag measurement out in a separate class as a singlton class or a static util. I cannot achieve it today with today's check is PCS does not have information of a few version level configs so I want to add that support as well. After that I think we can migrate that out and gradually make SIT thinner

Reply:
Actually, I changed it bad. This change was part of the failed refactor I did locally for the above reason, but as described in the PR, I want to make incremental, reviewable progress so I destroyed that part of changes. 
Changed it back now, will be in the next push.